registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/cmdvalidator/DeviceValidator.java

```diff
@@ -0,0 +1,142 @@
+	private String deviceValidateHistoryId;
+
+	/**
+	 * Checks if is device active.
+	 *
+	 * @param regOsi                the regOsi dto
+	 * @param registrationStatusDto
+	 * @throws IOException
+	 * @throws JsonProcessingException
+	 * @throws BaseCheckedException,                               ApisResourceAccessException
+	 * @throws com.fasterxml.jackson.databind.JsonMappingException
+	 * @throws com.fasterxml.jackson.core.JsonParseException
+	 */
+
+	public void validate(RegOsiDto regOsi, String registrationId)
+			throws JsonProcessingException, IOException, BaseCheckedException, ApisResourceAccessException {
+
+		List<NewRegisteredDevice> registreredDevices = regOsi.getCapturedRegisteredDevices();
+		if (registreredDevices != null && !registreredDevices.isEmpty()) {
+			for (NewRegisteredDevice deviceDetails : registreredDevices) {
+				RegisteredDevice registeredDevice = convert(deviceDetails);
+				DeviceValidateHistoryRequest deviceValidateHistoryRequest = new DeviceValidateHistoryRequest();
+				deviceValidateHistoryRequest.setDeviceCode(registeredDevice.getDeviceCode());
+				deviceValidateHistoryRequest.setDeviceServiceVersion(registeredDevice.getDeviceServiceVersion());
+				deviceValidateHistoryRequest.setDigitalId(registeredDevice.getDigitalId());
+				deviceValidateHistoryRequest.setTimeStamp(regOsi.getPacketCreationDate());
+				RequestWrapper<DeviceValidateHistoryRequest> request = new RequestWrapper<>();
+
+				request.setRequest(deviceValidateHistoryRequest);
+				request.setId(deviceValidateHistoryId);
+				request.setMetadata(null);
+				request.setVersion("1.0");
+				DateTimeFormatter format = DateTimeFormatter.ofPattern(env.getProperty(DATETIME_PATTERN));
+				LocalDateTime localdatetime = LocalDateTime
+						.parse(DateUtils.getUTCCurrentDateTimeString(env.getProperty(DATETIME_PATTERN)), format);
+				request.setRequesttime(localdatetime);
+
+				DeviceValidateHistoryResponse deviceValidateResponse;
+
+				ResponseWrapper<?> responseWrapper = (ResponseWrapper<?>) registrationProcessorRestService
+						.postApi(ApiName.DEVICEVALIDATEHISTORY, "", "", request, ResponseWrapper.class);
+				deviceValidateResponse = mapper.readValue(mapper.writeValueAsString(responseWrapper.getResponse()),
+						DeviceValidateHistoryResponse.class);
+				if (responseWrapper.getErrors() == null || responseWrapper.getErrors().isEmpty()) {
+					if (!deviceValidateResponse.getStatus().equalsIgnoreCase(VALID)) {
+						throw new ValidationFailedException(StatusUtil.DEVICE_VALIDATION_FAILED.getMessage(),
+								StatusUtil.DEVICE_VALIDATION_FAILED.getCode());
+					}
+				} else {
+					List<ErrorDTO> error = responseWrapper.getErrors();
+					regProcLogger.error("validate call ended for registrationId {} with error data : {}",
+							registrationId, error.get(0).getMessage());
+					throw new BaseCheckedException(
+							error.get(0).getMessage() + " " + "for" + " " + deviceDetails.getDigitalId().getType(),
```

Comment:
Replace " " + "for" + " "    with " for " . Try avoiding unwanted concatinations for static strings.

Reply:
resolved